src/DatesSimple.java

```diff
@@ -0,0 +1,52 @@
+public class DatesSimple {
+    public static void main(String[] args) {
+        int fromy = 2000, fromm = 2, fromd = 3;
+        int toy = 2127, tom = 11, tod = 29;
+
+        System.out.println("**** Original\n" +
+                "From " + fromy + "/" +
+                fromm + "/" + fromd + " to " +
+                toy + "/" + tom + "/" + tod);
+//      pack
+        int period = 0;
+        int resetFromy = fromy - 2000; // adjust the fromy to be in range [0, 127]
+        int resetToy = toy - 2000; // adjust the toy to be in range [0, 127]
+//      pack the "from" date (fromy/m/d) into a 16 bits integer
+/*
+      - Save variable "resetFromy" (range 0-127, requiring 7 bits, as 2^7 = 128)
+        into bits 9-15 of packedDate by using (resetFromy & 0x7f) << 9)
+      - Save variable "fromm" (range 0-15, requiring 4 bits, as 2^4 = 16)
+        into bits 5-8 of packedDate by using (fromm & 0x0f) << 5
+      - Save variable "fromd" (range 0-31, requiring 5 bits, as 2^5 = 32)
+        into bits 0-4 of packedDate by using (fromd & 0x1f)
+*/
+        int packedDate = (resetFromy & 0x7f) << 9 | (fromm & 0x0f) << 5 | (fromd & 0x1f);
+//      shift the packedDate to the left 16 bits to be at higher bits
+/*
+      - What happens after this shift:
+        So far the "from" date is stored on bits 0-15 of packeDate.
+        After left shifting the packedDate by 16 bits, the "from" date is stored on bits 15-31.
+      - Why am I doing it:
+        Because I want to store the "to" date on bits 0-15
```

Comment:
Please move the answer to the why question above the answer to what question. 

In general, why? is more important than what?